Add a Shimmer effect

bwhtmn opened the pull request:
Add an animated shimmer effect.
Resolves #33.

https://github.com/twostraws/Inferno/assets/1206592/c46615e2-5012-4574-beb1-7307c647738c

Created the shimmering by modifying lightness rather than alpha as in [SwiftUI-Shimmer](https://github.com/markiv/SwiftUI-Shimmer) that was linked to in the issue. This lightness approach was used since I found that non-opaque alpha values applied in a shader don't show the background views, probably because the shader is applied in the final rendering pass.

twostraws commented on Sources/Inferno/Shaders/Transformation/Shimmer.metal:
```diff
@@ -0,0 +1,119 @@
+//  Shimmer.metal
+//  Inferno
+//
+//  Created by Brent Whitman on 2024-04-10.
```
Please copy in the same header comments as the other files; it directs folks to the repository and license.

twostraws commented on README.md:
```diff
@@ -1601,6 +1653,7 @@ Some shaders were ported to Metal by me, from other open-source samples also rel
 - Circle, Circle Wave, Diamond, Diamond Wave are based on [Polka Dots Curtain](https://gl-transitions.com/editor/PolkaDotsCurtain) by bobylito.
 - Crosswarp is based on [Crosswarp](https://gl-transitions.com/editor/crosswarp) by Eke Péter.
 - Radial is based on [Radial](https://gl-transitions.com/editor/Radial) by Xaychru / gre.
+- Shimmer is inspired by [SwiftUI-Shimmer](https://github.com/markiv/SwiftUI-Shimmer) by markiv.
```
I would add your name here too; this is the correct place for credits.

bwhtmn commented:
> Great job – thank you! I've added two small comments, but otherwise this can be merged.

Thanks for the quick review! 😄 I've addressed both of your comments.

twostraws commented:
Perfect! 🙌
